docs/src/tutorials/child-device-config-mngmnt.md

```diff
@@ -0,0 +1,297 @@
+
+```
+files = [
+    { path = '/path/to/some/config', type = 'config1'},
+]
+```
+* `path` is the full path to the configuration file on the child device file system.
+* `type` is a unique alias for each file entry which will be used to represent that file in Cumulocity IoT.
+
+Example:
+
+```
+files = [
+    { path = '/home/pi/config/config1', type = 'config1'},
+    { path = '/home/pi/config/config2', type = 'config2'},
+]
+```
+Note: these referenced files including the path should be created.
+
+2. Upload this file to thin-edge.io via HTTP: The child device agent needs to upload this file to thin-edge.io with an HTTP PUT request to the URL:
+
+`
+http://{tedge-ip}:8000/tedge/file-transfer/{child-id}/c8y-configuration-plugin
+`
+
+* `{tedge-ip}` is the IP of the thin-edge.io device which is configured as `mqtt.external.bind_address` or `mqtt.bind_address` or `127.0.0.1` if neither is configured.
+* `{child-id}` is the child-device-id
+
+
+
+example: `curl -X PUT --data-binary @/home/pi/config/c8y-configuration-plugin.toml http://127.0.0.1:8000/tedge/file-transfer/child1/c8y-configuration-plugin
+`
+
+3. Notify thin-edge.io about the upload via MQTT
+
+
+Once the upload is complete, the agent should notify thin-edge.io about the upload by sending the following MQTT message:
+
+Topic:
+
+`tedge/{child-d}/commands/res/config_snapshot`
+
+Payload:
+
+`{ "type": "c8y-configuration-plugin”, "path": ”/child/local/fs/path”}`
+
+Example:
+
+Note: For using the `mosquito-pub` command (and later on in this tutorial also the `mosquitto_sub` command), you need to install the  mosquitto clients via `sudo apt-get install mosquitto-clients`
+
+`mosquitto_pub -h 127.0.0.1 -t "tedge/child1/commands/res/config_snapshot" -m '{"status": null,  "path": "", "type":"c8y-configuration-plugin", "reason": null}'
+`
+
+
+### Step 2 Handle config snapshot request from thin-edge.io
+
+In this step configuration files from a child device will be requested to make them visible in Cumulocity IoT.
+
+Handling config snapshot requests from thin-edge.io is a 4-step process:
+
+1. Subscribe to, and receive config snapshot requests via MQTT: `mosquitto_sub -h localhost -t "#"`
+2. In Cumulocity IoT, go to `Device Management` -> `All Devices` -> `Devices`, then click on your device. Then click on  `Child devices`  then click on the in *Step 1 Startup / Bootstrap of the child device* defined child device (child1) and then click on `Configuration`.
+3. On the right side, under configurations, click on the in *Step 1 Startup / Bootstrap of the child device* created config file (`config1`) in the column: `DEVICE-SUPPORTED CONFIGURATIONS` click on the config file (`config1`) and click on `Get snapshot from device`. This will trigger a config snapshot requests via MQTT.
```

Comment:
```suggestion
3. On the right side, under configurations, click on the config file (`config1`) created in *Step 1 Startup / Bootstrap of the child device* in the column: `DEVICE-SUPPORTED CONFIGURATIONS` and then click on `Get snapshot from device`. This will trigger a config snapshot request to the child device via MQTT.
```
An image here with those buttons highlighted would be nice.